### PR TITLE
Documentation overhaul

### PR DESCRIPTION
**Pull request by nevkontakte**
- Added a detailed documentation page about all things compatibility.
  - Updated supported packages list.
  - Updated README for 1.16.

I also tried to proofread myself, but it's unlikely that I've caught everything 🙃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,27 +7,34 @@ GopherJS - A compiler from Go to JavaScript
 
 GopherJS compiles Go code ([golang.org](https://golang.org/)) to pure JavaScript code. Its main purpose is to give you the opportunity to write front-end code in Go which will still run in all browsers.
 
+### What's new?
+
+ - 2021-04-04: **Go 1.16 is now officially supported!** 🎉 🎉 🎉
```
Comment (flimzy) [inline README.md:12]: :tada:

File: README.md [hunk 2]
```diff
@@ -7,27 +7,34 @@ GopherJS - A compiler from Go to JavaScript
...
 
 ### What is supported?
-Nearly everything, including Goroutines ([compatibility table](https://github.com/gopherjs/gopherjs/blob/master/doc/packages.md)). Performance is quite good in most cases, see [HTML5 game engine benchmark](https://ajhager.github.io/engi/demos/botmark.html). Cgo is not supported.
+
+Nearly everything, including Goroutines ([compatibility documentation](https://github.com/gopherjs/gopherjs/blob/master/doc/compatibility.md)). Performance is quite good in most cases, see [HTML5 game engine benchmark](https://ajhager.github.io/engi/demos/botmark.html). Cgo is not supported.
 
 ### Installation and Usage
-GopherJS requires Go 1.12 or newer.
+
+GopherJS requires Go 1.16 or newer.
```
Comment (flimzy) [inline README.md:24]: Do we want to add a sentence about using an old version if you require something older?  Maybe with a link to the GitHub tags or releases page?

File: doc/compatibility.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    - Limited [compiler directive](pragma.md) (a.k.a. "pragma") support. Those are considered compiler implementation-specific and are generally not portable.
+    - GopherJS ships with [standard library augmentations](../compiler/natives/src/), that are required to make it work in a browser. Those are applied on-the-fly during the build process and are generally invisible to any third-party tooling such as linters. In most cases that shouldn't matter, since they never change public interfaces of the standard library packages, but this is something to be aware of.
+    - Runtime debuggers and profilers. Since GopherJS compiles Go to JavaScript, one must use JavaScript debuggers and profilers (e.g. browser dev tools) instead of the normal Go ones (e.g. delve or pprof). Unfortunately, limited sourcemap support makes this experience less than ideal at the moment.
+
+## Go version compatibility
+
+In general, for a given release of GopherJS the following statements _should_ be true:
+
+  - GopherJS compiler can be built from source with the Go latest stable Go release at the time when the GopherJS release is created, or any newer Go release.
```
Comment (flimzy) [inline doc/compatibility.md:18]:
```suggestion
  - GopherJS compiler can be built from source with the latest stable Go release at the time when the GopherJS release is created, or any newer Go release. 
```

File: README.md (hunk 2 above)
Comment (nevkontakte) [reply README.md:24]: Added a link to the explanation in docs/compatibility.md and a lint to GitHub Releases.

File: doc/compatibility.md (hunk 3 above)
Comment (nevkontakte) [reply doc/compatibility.md]: Done.

**Pull request merged**